loopin.go

```diff
@@ -582,16 +601,18 @@ func (s *loopInSwap) publishTimeoutTx(ctx context.Context,
 
 // persistState updates the swap state and sends out an update notification.
 func (s *loopInSwap) persistState(ctx context.Context) error {
-	updateTime := time.Now()
-
-	s.lastUpdateTime = updateTime
-
 	// Update state in store.
-	err := s.store.UpdateLoopIn(s.hash, updateTime, s.state)
+	err := s.store.UpdateLoopIn(s.hash, s.lastUpdateTime, s.state)
 	if err != nil {
 		return err
 	}
 
 	// Send out swap update
 	return s.sendUpdate(ctx)
 }
+
+// setState updates the swap state and last update timestamp.
```

Comment:
I think we should also roll in `persistState` into this method as well. Otherwise, I see a few instances above, where we modify the swap state, but don't persist it to the database.

Reply:
Scratch that, I see it's persisted for the first thing that causes it to break out of the loop.